ToolBox/AlgoSeekOptionsConverter/Program.cs

```diff
@@ -29,72 +32,32 @@ public class Program
 
         static void Main()
         {
-            var directory = "F:/Downloads/AlgoSeek/smaller";
-            directory = "F:/AlgoSeek/20151224";
-            var dataDirectory = "./Data";
+			Environment.SetEnvironmentVariable("MONO_MANAGED_WATCHER", "disabled");
+
+			var sourceDirectory = "/cache/TestData/RawGoog/20151224";
+			var destinationDirectory = "/cache/TestData/Converted";
+
+//			var sourceDirectory = "/cache/DevData/RawDataSmall/20140516";
+//			var destinationDirectory = "/cache/DevData/ConvertedData";
 
             Log.LogHandler = new CompositeLogHandler(new ILogHandler[]
             {
                 new ConsoleLogHandler(),
                 new FileLogHandler(LogFilePath)
             });
 
-            // first process tick/second/minute -- we'll do hour/daily at the end on a per symbol basis
-            var parallelism = 4;
-            var options = new ParallelOptions {MaxDegreeOfParallelism = parallelism};
-            var resolutions = new[] {Resolution.Tick, Resolution.Minute, Resolution.Second, Resolution.Hour, Resolution.Daily};
-            
-            // for testing only process the smallest 2 files
-            var files = Directory.EnumerateFiles(directory).OrderByDescending(x => new FileInfo(x).Length);
-            Parallel.ForEach(files, options, file =>
-            {
-                Log.Trace("Begin tick/second/minute/hour/daily: " + file);
-
-                var quotes = DataProcessor.Zip(dataDirectory, resolutions, TickType.Quote, true);
-                var trades = DataProcessor.Zip(dataDirectory, resolutions, TickType.Trade, true);
-
-                var streamProvider = StreamProvider.ForExtension(Path.GetExtension(file));
-                if (!RawFileProcessor.Run(file, new[] {file}, streamProvider, new AlgoSeekOptionsParser(), quotes, trades))
-                { 
-                    return;
-                }
-
-                Log.Trace("Completed tick/second/minute/hour/daily: " + file);
-            });
-            
-            Log.Trace("Begin compressing csv files");
-
-            var root = Path.Combine(dataDirectory, "option", "usa");
-            var fine =
-                from res in new[] {Resolution.Tick, Resolution.Second, Resolution.Minute}
-                let path = Path.Combine(root, res.ToLower())
-                from sym in Directory.EnumerateDirectories(path)
-                from dir in Directory.EnumerateDirectories(sym)
-                select new DirectoryInfo(dir).FullName;
+			var flushInterval = 1000000L;
 
-            var coarse =
-                from res in new[] {Resolution.Hour, Resolution.Daily}
-                let path = Path.Combine(root, res.ToLower())
-                from dir in Directory.EnumerateDirectories(path)
-                select new DirectoryInfo(dir).FullName;
+			Stopwatch sw = Stopwatch.StartNew();
+			AlgoSeekOptionsConverter.ConvertToFineResolution(sourceDirectory, destinationDirectory, flushInterval);
```

Comment:
var stopwatch, no contractions where possible. Datetime would also suffice since its running for a long time.